Gems/EMotionFX/Code/Source/Integration/Editor/Components/EditorActorComponent.cpp

```diff
@@ -541,7 +541,12 @@ namespace EMotionFX
             m_actorAsset = asset;
             AZ_Assert(m_actorAsset.IsReady() && m_actorAsset->GetActor(), "Actor asset should be loaded and actor valid.");
 
-            CheckActorCreation();
+            // We'll defer actor creation until the next tick on the tick bus. This is because OnAssetReady() can sometimes get
+            // triggered while in the middle of the render tick, since the rendering system sometimes contains blocking loads
+            // which will still process any pending OnAssetReady() commands while waiting. If that occurs, the actor creation
+            // would generate errors from trying to create a rendering actor while in the middle of processing the rendering data.
+            // We can avoid the problem by just always waiting until the next tick to create the actor.
+            m_processLoadedAsset = true;
```

Comment:
Another option is to enqueue it on the tick bus rather than having a member var, but it would be bad if you enqueue it multiple times, so probably this is safer since I don't think we have an EnqueueOnce() function.

Reply:
There are a couple of reasons not to do it that way:
1. This component literally already responds to OnTick(), so there's no value in enqueueing a second method onto the tick bus for the same component that could fire in either order. It's just more overhead and confusion for no gain.
2. (The main reason) If the component gets destroyed before the next tick, it's actually really hard to write a function that correctly does what it needs to for the component without getting tripped up by the lifetime issue, since it can't use a "this" pointer inside of it. On the flip side, as you can see in the PR, it's super _easy_ to get the lifetime management correct by having the component attach/unattach to the TickBus, because then it simply won't run the queued processing if the component gets destroyed before the queued work needs to happen.